webapp/channels/src/components/suggestion/search_date_suggestion/search_date_suggestion.tsx

```diff
@@ -7,15 +7,22 @@ import {DayPicker} from 'react-day-picker';
 
 import type {Locale} from 'date-fns';
 
-import Suggestion from '../suggestion.jsx';
+import {SuggestionProps} from '../suggestion';
 
 import * as Keyboard from 'utils/keyboard';
 import * as Utils from 'utils/utils';
 import Constants from 'utils/constants';
 
 import 'react-day-picker/dist/style.css';
 
-export default class SearchDateSuggestion extends Suggestion {
+type Props = SuggestionProps<never> & {
+    currentDate?: Date;
+    handleEscape: () => void;
+    locale: string;
+    preventClose: () => void;
+}
+
+export default class SearchDateSuggestion extends React.PureComponent<Props> {
```

Comment:
This one was never a true `Suggestion` component, and it has more internal logic, so I didn't bother making it functional. It kind of abuses the suggestion logic because it appears in the search box the same way `from:` shows an autocomplete list